tests/unittest/_torch/attention/sparse/test_dsa_indexer.py

```
@@ -476,6 +473,177 @@ def __init__(self):
+        )
+
+        # Show first few differences
+        diff_indices = torch.nonzero(diff_mask.view(-1))[:5]
+        for idx in diff_indices:
+            flat_idx = idx.item()
+            print(
+                f"  Byte {flat_idx}: CUDA={k_cache_cuda.view(-1)[flat_idx].item()}, "
+                f"Python={k_cache_python.view(-1)[flat_idx].item()}")
+
+        # Fail the test
+        assert False, f"CUDA kernel produced different results than Python reference"
```

Ruff (0.14.3)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)